server/src/main/java/org/elasticsearch/search/builder/SearchSourceBuilder.java

```diff
@@ -1485,6 +1553,23 @@ private SearchSourceBuilder parseXContent(XContentParser parser, boolean checkTr
                         }
                     }
                     searchUsage.trackSectionUsage(KNN_FIELD.getPreferredName());
+                } else if (SUB_SEARCHES_FIELD.match(currentFieldName, parser.getDeprecationHandler())) {
+                    if (subSearchSourceBuilders.isEmpty() == false) {
+                        throw new IllegalArgumentException(
+                            "cannot specify field [" + currentFieldName + "] and field [" + QUERY_FIELD.getPreferredName() + "]"
+                        );
+                    }
+                    while ((token = parser.nextToken()) != XContentParser.Token.END_ARRAY) {
+                        if (token == XContentParser.Token.START_OBJECT) {
+                            subSearchSourceBuilders.add(SubSearchSourceBuilder.fromXContent(parser, searchUsage));
+                        } else {
+                            throw new XContentParseException(
+                                parser.getTokenLocation(),
+                                "malformed query within the [sub_searches] field; found " + token
+                            );
+                        }
+                    }
+                    searchUsage.trackSectionUsage(SUB_SEARCHES_FIELD.getPreferredName());
```

Comment:
there's quite some API other than search that parse part of their content into a SearchSourceBuilder. Is it ok that they all end up supporting rank and sub_searches automatically?

Reply:
I think it should work for any API that wants to use the search endpoint, but it's possible I'm missing something.